### ISSUE TITLE
AWS SES: Email address encoding issue (+ (plus) addresses unsupported)

### ISSUE THREAD

**Issue by marcus233** [n8n-io/n8n#9794]
### Bug Description

The AWS SES node gives an error if the sender or receiver use + (plus) address. This type of address is supported SES (The email address string must be 7-bit ASCII [1]) and can be verified by sending test email with to/from + addresses in the AWS Console. 

In n8n  when sending either to a + address, or sending from a + address the following error occurs:
```
400 - "<ErrorResponse xmlns=\"http://ses.amazonaws.com/doc/2010-12-01/\">
 <Error>
 <Type>Sender</Type>
 <Code>InvalidParameterValue</Code>
 <Message>Local address contains control or whitespace</Message>
 </Error>
 <RequestId>a06f54ba-e4da-4af9-9477-d2247f3154dd</RequestId>
</ErrorResponse>"
```
with the traceback: 
```
NodeApiError: Bad request - please check your parameters
    at Object.requestWithAuthentication (/usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/NodeExecuteFunctions.js:1266:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.requestWithAuthentication (/usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/NodeExecuteFunctions.js:1994:20)
    at Object.awsApiRequest (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/SES/GenericFunctions.js:24:16)
    at Object.awsApiRequestSOAP (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/SES/GenericFunctions.js:42:22)
    at Object.execute (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/SES/AwsSes.node.js:925:40)
    at Workflow.runNode (/usr/local/lib/node_modules/n8n/node_modules/n8n-workflow/dist/Workflow.js:728:19)
    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/WorkflowExecute.js:664:51
    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/WorkflowExecute.js:1079:20
```

This is a problem because we are accepting user input for the receiving address and they may use a + address for their own filtering rules. We are also using a + address as the sender so we can later match unique email conversations from replies.

1. https://docs.aws.amazon.com/ses/latest/dg/send-email-raw.html#send-email-mime-encoding-addresses

### To Reproduce

1. Create an SES node
2a. Try sending an email me@home.com from someone+something@somewhere.com  (+ address in sender)
2b. Try sending an email to me+test@home.com from someone@somewhere.com (+ address in receiver)

### Expected behavior

n8n should allow sender and receiver addresses with + in them.

### Operating System

Alpine (n8n's docker image)

### n8n Version

1.45.0 (Self Hosted)

### Node.js Version

20 (n8n's docker image)

### Database

PostgreSQL

### Execution mode

queue

**Comment by Joffcom** [comment]
Hey @marcus233,

Thanks for the report, The error coming back is Amazon saying "Local address contains control or whitespace" 

This is not us creating the response but I suspect we are not encoding the value before sending it which could be the issue.

I will see if I can reproduce this in the morning.

**Comment by Joffcom** [comment]
Hey @marcus233,

I have managed to reproduce and have put in a fix, Once reviewed it will be merged and available in a future release.

**Comment by marcus233** [comment]
> I have managed to reproduce and have put in a fix, Once reviewed it will be merged and available in a future release.

Thank you! Thanks for the quick fix.

**Comment by marcus233** [comment]
@Joffcom we tested your fix on the 1.47 pre-release and it solves our issue. Thanks again!

**Comment by Joffcom** [comment]
Hey @marcus233,

Thanks for the update I will mark this as closed.
